HTTPBearer token is set, Auth button not shown on /api/docs

Igoranze opened the pull request:
Part of https://github.com/workfloworchestrator/orchestrator-core/issues/670

This PR adds a Class TokenExtractor that is used as a Depends and extracts the token from the request in order to allow for HTTPBearer requests.
Part of the missing HTTPBearer request was that the /api/docs auth button was not shown.

Also added a statement to the authentication in order to filter out strawberry requests and get the token via the HTTPBearer without the Depends

pboers1988 commented:
@Igoranze still needs some linting fixes

pboers1988 reviewed (changes requested):
Linting!

torkashvand commented:
I think linting (mypy) errors here are really important. as fixing them, more changes reveals

Igoranze commented:
> I think linting (the mypy) errors here are really important. as fixing them, more changes reveals for example, we used the border type `HTTPConnection` because we wanted to satisfy websocket requests as well

@torkashvand I have changed the request type to `HTTPConnection` in the `TokenExtractor`, in the `authenticate` method I changed it to `Request` since websocket requests are are checkend in the `TokenExtractor` this would compute. What are your findings on this?

Mark90 reviewed (commented):
LGTM! 2 small comments, and can you bump the version from 2.2.0 to 2.3.0?

Mark90 commented:
@pboers1988 let's briefly wait with merging this, I want to run it through some tests in okteto on thursday

Mark90 commented on oauth2_lib/fastapi.py:
```diff
@@ -168,11 +175,7 @@ def __init__(
         resource_server_id: str,
         resource_server_secret: str,
         oidc_user_model_cls: type[OIDCUserModel],
-        id_token_extractor: IdTokenExtractor | None = None,
     ):
-        if not id_token_extractor:
-            self.id_token_extractor = HttpBearerExtractor()
```
Maybe keep the kwarg in `__init__` and when a value is passed log a (deprecation) warning instead?

Mark90 replied:
Great spot by you (@Igoranze): if a custom `id_token_extractor` is passed then it's not actually assigned to `self.id_token_extractor` so `authenticate()` raises an `AttributeError` when trying to call it.

The only way this can have worked for others is if they override `authenticate()` as well. So, we don't need to worry about backwards compatibility

Mark90 reviewed (approved):
Nice :)

torkashvand commented on oauth2_lib/fastapi.py:
```diff
@@ -197,33 +200,16 @@ async def authenticate(self, request: HTTPConnection, token: str | None = None)
-                    if not extracted_id_token:
-                        raise HTTPException(status_code=HTTP_403_FORBIDDEN, detail="Not authenticated")
+        if not token:
+            raise HTTPException(status_code=HTTP_403_FORBIDDEN, detail="Not authenticated")
```
I think you did this because the new WebSocket implementation passes the token in a header. Nice!

Igoranze replied:
Correct :) And it no longer servers only for the websocket since that should not matter
